src/anomalib/models/image/supersimplenet/torch_model.py

```diff
@@ -0,0 +1,372 @@
+            feat_h (int): feature height.
+            feat_w (int): feature width.
+
+        Returns:
+            (torch.Tensor): downsampled masks.
+        """
+        masks = masks.type(torch.float32)
+        # best downsampling proposed by DestSeg
+        masks = F.interpolate(
+            masks.unsqueeze(1),
+            size=(feat_h, feat_w),
+            mode="bilinear",
+        )
+        return torch.where(
+            masks < 0.5,
+            torch.zeros_like(masks),
+            torch.ones_like(masks),
+        )
+
+
+def init_weights(m: nn.Module) -> None:
+    """Init weight of the model.
+
+    Args:
+        m (nn.Module): torch module.
+    """
+    if isinstance(m, nn.Linear | nn.Conv2d):
+        nn.init.xavier_normal_(m.weight)
+    elif isinstance(m, nn.BatchNorm1d | nn.BatchNorm2d):
+        nn.init.constant_(m.weight, 1)
+
+
+class FeatureExtractor(nn.Module):
+    """Feature extractor module.
+
+    Args:
+        backbone (str): backbone name.
+        layers (list[str]): list of layers used for extraction.
+    """
+
+    def __init__(self, backbone: str, layers: list[str], patch_size: int = 3) -> None:
+        super().__init__()
+
+        self.feature_extractor = TorchFXFeatureExtractor(
+            backbone=backbone,
+            return_nodes=layers,
+            weights="IMAGENET1K_V1",
+        )
+        self.pooler = nn.AvgPool2d(
+            kernel_size=patch_size,
+            stride=1,
+            padding=patch_size // 2,
+        )
+
+    def forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
+        """Extract features from input tensor.
+
+        Args:
+            input_tensor: input tensor (images)
+
+        Returns:
+            (torch.Tensor): extracted feature map.
+        """
+        # extract features
+        self.feature_extractor.eval()
+        with torch.no_grad():
+            features = self.feature_extractor(input_tensor)
+
+        features = list(features.values())
+
+        _, _, h, w = features[0].shape
+        feature_map = []
+        for layer in features:
+            # upscale all to 2x the size of the first (largest)
+            resized = F.interpolate(
+                layer,
+                size=(h * 2, w * 2),
+                mode="bilinear",
+            )
+            feature_map.append(resized)
+        # channel-wise concat
+        feature_map = torch.cat(feature_map, dim=1)
+
+        # neighboring patch aggregation
+        return self.pooler(feature_map)
+
+    def get_channels_dim(self) -> int:
+        """Get feature channel dimension.
+
+        Returns:
+            (int): feature channel dimension.
+        """
+        # dryrun
+        self.feature_extractor.eval()
+        with torch.no_grad():
+            features = self.feature_extractor(torch.rand(1, 3, 256, 256))
+        # sum channels
+        return sum(feature.shape[1] for feature in features.values())
+
+
+class FeatureAdaptor(nn.Module):
```

Comment:
I would prefer `FeatureAdapter`, since we try to use American English throughout the code base. but I guess it would cause a discrepancy with the official implementation in your own repo. 

@samet-akcay @ashwinvaidya17 thoughts?

Reply:
as someone based in the UK, not sure what to say here 😁

Reply:
I changed this to FeatureAdapter, I think it's clear enough when compared with the official code 😄